.eslintrc.js

```diff
@@ -23,26 +23,102 @@ const globalVars = parse(babelParse(data, { sourceType: 'module' }))
   );
 
 module.exports = {
-  extends: 'rainbow',
+  env: {
+    es2022: true,
+    browser: true,
+    node: true,
+  },
+
+  parser: '@typescript-eslint/parser',
+
+  parserOptions: {
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true,
+    },
+    project: ['./tsconfig.json'],
+  },
+
+  extends: [
+    'eslint:recommended',
+    'plugin:eslint-comments/recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'plugin:jest/recommended',
+
+    /**
+     * For eslint-plugin-import
+     *
+     * @see https://github.com/import-js/eslint-plugin-import/blob/d45fe21bfa09f61402c68c3d271250d95f9c9ed3/README.md?plain=1#L156
+     */
+    'plugin:import/recommended',
+    'plugin:import/typescript',
+
+    /**
+     * Last, must override thing
+     */
+    'prettier',
+  ],
+
+  globals: globalVars,
+
   settings: {
+    'react': {
+      version: 'detect',
+    },
     'import/resolver': {
-      'node': {
-        extensions: [
-          '.js',
-          '.ios.js',
-          '.android.js',
-          '.native.js',
-          '.ts',
-          '.tsx',
-        ],
-      },
-      'babel-module': {
-        alias: {},
-      },
+      'typescript': true,
+      'node': true,
+      'babel-module': true,
     },
   },
-  plugins: [],
-  globals: globalVars,
-  rules: {},
-  env: { browser: true, node: true },
+
+  rules: {
+    // TODO can fix all of these
+    'no-var': 'warn',
+    'prefer-const': 'warn',
+    'prefer-rest-params': 'warn',
+    'prefer-spread': 'warn',
+    'no-async-promise-executor': 'warn',
+    'no-prototype-builtins': 'warn', // this one is probably fine
+    'no-irregular-whitespace': 'warn', // not a big deal most of the time
+
+    // Some of these are OK while we transition to TS
+    '@typescript-eslint/ban-ts-comment': 'warn',
```

Comment:
Thought's on erroring altogether and only allowing for `@ts-expect-error`? I don't really see the benefit of warning on `@ts-expect-error` as I think it would provide a false assumption that it's not recommended, and also `@ts-expect-error` ends up becoming a TS error if it's invalid (once the dependant line is migrated to TypeScript).

```suggestion
    '@typescript-eslint/ban-ts-comment': ['error', { 'ts-expect-error': 'allow-with-description' }],
```

Reply:
Yeah good shout! I think we should move this direction, but like the comment says, I figure we're OK with this for now since there are >400 `@ts-ignore` directives in the codebase post-TS migration. New code should for sure not use it, but not sure we can lint that and ignore old code 🤔

Reply:
Maybe `'warn'` and allow `@ts-expect-error` then?

Reply:
Ah yes of course, my bad!